src/metabase/models/channel.clj

```diff
@@ -43,10 +45,16 @@
 
 (t2/define-before-update :model/Channel
   [instance]
-  (assert-channel-type instance)
-  (when (false? (:active (t2/changes instance)))
-    (t2/delete! :model/PulseChannel :channel_id (:id instance)))
-  instance)
+  (let [deactivated? (false? (:active (t2/changes instance)))]
```

Comment:
nit: I think that `deactivation?` is a better name here, as it would be false for updates to an already deactivated record.